exonum/src/runtime/error.rs

```diff
@@ -456,6 +456,16 @@ impl ExecutionError {
         Self::new(ErrorKind::Service { code }, description)
     }
 
+    /// Creates a runtime error for use in runtime code.
+    pub fn runtime(runtime_id: u32, code: u8, description: impl Into<String>) -> Self {
```

Comment:
Funny, I had the same issue. You don't need this method, you can create runtime error using `ExecutionFail::with_description` method: 
`rust::Error::UnableToDeploy.with_description(description).into()`

@slowli perhaps the docs aren't clear enough, as even two developers have already stuck with it?

Reply:
I think asymmetry with `ExecutionError::service` does the thing here.

Reply:
Yep, I agree, I think that was the main reason for me. Sorry for requesting changes @Kormick , I pressed wrong button actually. I'm pretty okay with adding this method (and I've actually created a task to do so, but closed it when found another solution).

Reply:
Though using `ExecutionFail` trait is a bit better because the interface is much safer in this case - you're not forced to use numeric IDs (even if you convert them from enum)

Reply:
Can we provide runtime id with it?

Reply:
Good catch, actually not. 🤔 With `runtime` method, I don't see much sense in using `ExecutionFail` for runtime errors, because having runtime ID is useful. Which means we need either to add runtime ID to `ExecutionFail` somehow, or simplify `ExecutionFail` derive macros and use it for service errors only.
Please create a task for whatever decision. 

cc @slowli @aleksuss

Reply:
> Can we provide runtime id with it?

The runtime ID should be added automatically during error handling within the dispatcher; see, e.g. [here](https://github.com/exonum/exonum/blob/1a23a5c6bcb9c349ea096818e5cf952dc6669261/exonum/src/runtime/dispatcher/mod.rs#L321). If this information is not set, this should be fixed within the dispatcher.

Reply:
set_runtime_id method is module-private, we can't use it in EJB.

Reply:
I've added set of runtime_id in the Dispatcher. Should we leave ExecutionError::runtime method?

Reply:
@vitvakatu That's the idea behind automatic assignment of information - the runtimes should not care about this stuff knowing that it happens automatically behind the scenes.

@Kormick I don't see a reason to retain this method. I'd rather think about removing `service` constructor 🙃